src/caikit_nlp_client/http_client.py

```diff
@@ -242,10 +256,19 @@ def generate_text_stream(
                 continue
 
             buffer.clear()
+            if "details" in message and "code" in message:
+                raise RuntimeError(
+                    "Exception iterating responses: {}".format(message["details"])
+                )
             yield message["generated_text"]
```

Comment:
I just realized that we aren't checking the case in which the `generated_text` key is missing, let's add something like this to avoid exploding with a KeyError:
```suggestion
                )
            try:
                yield message["generated_text"]
            except KeyError:
                raise RuntimeError("Unexpected response from the server")
```

Reply:
Ok will do.